mne/io/persyst/persyst.py

```diff
@@ -0,0 +1,443 @@
+                meas_date = datetime(
+                    year=testdate.year, month=testdate.month,
+                    day=testdate.day, hour=testtime.hour,
+                    minute=testtime.minute, second=testtime.second,
+                    tzinfo=timezone.utc)
+
+        # Create mne structure
+        ch_names = list(channelmap_dict.keys())
+        if n_chs != len(ch_names):
+            raise RuntimeError('Channels in lay file do not '
+                               'match the number of channels '
+                               'in the .dat file.')  # noqa
+        # get rid of the "-Ref" in channel names
+        ch_names = [ch.split('-ref')[0] for ch in ch_names]
+
+        # get the sampling rate and default channel types to EEG
+        sfreq = fileinfo_dict.get('samplingrate')
+        ch_types = 'eeg'
+        info = create_info(ch_names, sfreq, ch_types=ch_types)
+        info.update(subject_info=subject_info)
+        for idx in range(n_chs):
+            info['chs'][idx]['cal'] = cal
+        info['meas_date'] = meas_date
+
+        # determine number of samples in file
+        # Note: We do not use the lay file to do this
+        # because clips in time may be generated by Persyst that
+        # DO NOT modify the "SampleTimes" section
+        with open(dat_fpath, 'rb') as f:
+            # determine the precision
+            if int(datatype) == 7:
+                # 32 bit
+                dtype_bytes = _fmt_byte_dict['long']
+            elif int(datatype) == 0:
+                # 16 bit
+                dtype_bytes = _fmt_byte_dict['short']
+
+            # allow offset to occur
+            f.seek(0, os.SEEK_END)
+            n_samples = f.tell()
+            n_samples = n_samples // (dtype_bytes * n_chs)
+
+            if verbose:
+                print(f'Loaded {n_samples} samples '
+                      f'for {n_chs} channels.')
+
+        raw_extras = {
+            'datatype': datatype,
+            'n_chs': n_chs,
+            'n_samples': n_samples
+        }
+
+        # create Raw object
+        super(RawPersyst, self).__init__(
+            info, preload, filenames=[dat_fpath],
+            last_samps=[n_samples - 1],
+            raw_extras=[raw_extras], verbose=verbose)
+
+        # set annotations based on the comments read in
+        num_comments = len(comments_dict.keys())
+        onset = np.zeros(num_comments, float)
+        duration = np.zeros(num_comments, float)
+        description = [''] * num_comments
+        for t_idx, (_description, (_onset, _duration)) in \
+                enumerate(comments_dict.items()):
+            # extract the onset, duration, description to
+            # create an Annotations object
+            onset[t_idx] = _onset
+            duration[t_idx] = _duration
+            description[t_idx] = _description
+            annot = Annotations(onset, duration, description)
+            self.set_annotations(annot)
```

Comment:
set_annotation in a loop seem weird

Reply:
are you sure you don't want to do this at the end of the for loop? note that set_annotations flushed previous annotations

Reply:
Ah yes that makes sense. You're right. I think I might've indented at some point w/o realizing it.